### ISSUE TITLE
Rustdoc `doc_auto_cfg` incorrectly reporting an item with conditionally default type params as conditional

### ISSUE THREAD

**Issue by Kixunil** [rust-lang/rust#112059]
When a generic type parameter is conditionally default `doc_auto_cfg` reports whole item as conditional.

I tried this code:

```rust
#[cfg(not(feature = "std"))]                       
pub enum ReadError<Decode, Read> {                 
    /// Reading from a reader failed.              
    Read(Read),                                    
    /// Decoding the value failed.                 
    Decode(Decode),                                
}                                                  

#[cfg(feature = "std")]                            
pub enum ReadError<Decode, Read = std::io::Error> {
    /// Reading from a reader failed.              
    Read(Read),                                    
    /// Decoding the value failed.                 
    Decode(Decode),                                
}                                                  
```

I expected to see this happen: The resulting documentation says that the default parameter is only available with `std`

Instead, this happened: The resulting documentation says that `ReadError` is only available with `std`

### Meta

`rustc --version --verbose`:
```
rustc 1.72.0-nightly (1c53407e8 2023-05-28)
binary: rustc
commit-hash: 1c53407e8c7cc922d718bde61ca34f47b6d2120f
commit-date: 2023-05-28
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.4
```

**Comment by GuillaumeGomez** [comment]
The `cfg` is put on the item and `rustdoc` only knows about the currently available item (and its generics). This is far beyond what the feature is.

I think the solution here would be to allow to add `cfg` on generics if it's not already allowed, but that's a big change.

**Comment by Kixunil** [comment]
Yeah, I realize it may be too much to make perfect doc. But perhaps it'd be possible to at least not show "only available with `std`" if negation of the configuration produces an item of the same name?

**Comment by GuillaumeGomez** [comment]
We don't have this information. We only know about the currently available items unfortunately.
